python-lib/dku_io_utils/model_selection.py

```diff
@@ -136,12 +127,26 @@ def _get_best_model(self):
                 df = df[df[METRICS_DATASET.TARGET_COLUMN] == METRICS_DATASET.AGGREGATED_ROW]
 
             assert df[METRICS_DATASET.MODEL_COLUMN].nunique() == len(df.index), "More than one row per model"
-            model_label = df.loc[df[self.performance_metric].idxmin()][
+            model_label = df.loc[df[performance_metric].idxmin()][
                 METRICS_DATASET.MODEL_COLUMN
             ]  # or idxmax() if maximize metric
             assert model_label in available_models_labels, "Best model retrieved is not an available models"
         except Exception as e:
             raise ModelSelectionError(
-                f"Unable to find the best model of session '{self.session_name}' with the performance metric '{self.performance_metric}'. Full error: {e}"
+                f"Unable to find the best model of session '{self.session_name}' with the performance metric '{performance_metric}'. Full error: {e}"
             )
         return model_label
+
+    @staticmethod
+    def find_all_models_labels_from_folder(folder, session_path=None):
```

Comment:
In 14eede2, a warning is logged and an error is raised when the method is called from the ModelSelection class (not for the dynamic select because raising an error there does nothing).